lib/android/src/main/java/com/reactnativeldk/LdkModule.kt

```diff
@@ -772,14 +774,27 @@ class LdkModule(reactContext: ReactApplicationContext) : ReactContextBaseJavaMod
             return handleReject(promise, LdkErrors.invoice_payment_fail_must_not_specify_amount)
         }
 
-        val res = if (isZeroValueInvoice)
-            UtilMethods.pay_zero_value_invoice(invoice, amountSats.toLong() * 1000, Retry.timeout(timeoutSeconds.toLong()), channelManager) else
-            UtilMethods.pay_invoice(invoice, Retry.timeout(timeoutSeconds.toLong()), channelManager)
+        val paymentId = invoice.payment_hash()
+        val detailsRes = if (isZeroValueInvoice)
+            UtilMethods.payment_parameters_from_zero_amount_invoice(invoice, amountSats.toLong()) else
+            UtilMethods.payment_parameters_from_invoice(invoice)
+
+        if (!detailsRes.is_ok) {
+            return handleReject(promise, LdkErrors.invoice_payment_fail_invoice)
+        }
+
+        val sendDetails = detailsRes as Result_C3Tuple_ThirtyTwoBytesRecipientOnionFieldsRouteParametersZNoneZ_OK
+        val paymentHash = sendDetails.res._a
+        val recipientOnion = sendDetails.res._b
+        val routeParams = sendDetails.res._c
+
+        val res = channelManager!!.send_payment(paymentHash, recipientOnion, paymentId, routeParams, Retry.timeout(timeoutSeconds.toLong()))
+
         if (res.is_ok) {
             channelManagerPersister.persistPaymentSent(hashMapOf(
                 "bolt11_invoice" to paymentRequest,
                 "description" to (invoice.into_signed_raw().raw_invoice().description()?.into_inner() ?: ""),
```

Comment:
I noticed this needs an update as well, otherwise the invoice description is always `null`:
```kt
"description" to (invoice.into_signed_raw().raw_invoice().description()?.to_str() ?: ""),
```

Reply:
Nice thanks updated